python/pyspark/sql/functions/builtin.py

```diff
@@ -3671,16 +3671,84 @@ def regr_avgx(y: "ColumnOrName", x: "ColumnOrName") -> Column:
 
     Examples
     --------
-    >>> from pyspark.sql import functions as sf
-    >>> x = (sf.col("id") % 3).alias("x")
-    >>> y = (sf.randn(42) + x * 10).alias("y")
-    >>> spark.range(0, 1000, 1, 1).select(x, y).select(
-    ...     sf.regr_avgx("y", "x"), sf.avg("x")
-    ... ).show()
+    Example 1: All paris are non-null
+
+    >>> import pyspark.sql.functions as sf
+    >>> from pyspark.sql.types import IntegerType, StructField, StructType
+    >>> schema = StructType([
+    ...   StructField('y', IntegerType(), True),
+    ...   StructField('x', IntegerType(), True)
+    ... ])
+    >>> df = spark.createDataFrame([(1, 2), (2, 2), (2, 3), (2, 4)], schema)
```

Comment:
shall we simplify the dataframe creation (here and other places) by single line:
```
spark.sql("SELECT * FROM VALUES (1, 2), (2, NULL), (3, 3), (4, 4) AS t(x, y)")
```
or
```
spark.createDataFrame([(1, 2), (2, None), (2, 3), (2, 4)], ['x', 'y'])
```

Reply:
It seems the specified schema is not needed

Reply:
All suggestions received, let me optimize it.